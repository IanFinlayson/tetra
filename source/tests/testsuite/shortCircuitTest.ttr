def main():
   a = true
   print("In main(), just created a and a = ", a, "\n")
   b = false
   print("In main(), just created b and b = ", b, "\n")
   x = [0]
   print("In main(), just created x and x = ", x, "\n")
   c = " and true"
   print("In main(), just created c and c = ", c, "\n")
   d = " or true"
   print("In main(), just created d and d = ", d, "\n")
   if(a and boolTest(x)):
      print("a and booltest(x) are both true.", "\n")
      x[0] += 2
      print("In main(), just added 2 to x[0] and x = ", x, "\n")
   if(a or boolTest(x)):
      print("a or booltest(x) is true.", "\n")
      x[0] += 2
      print("In main(), just added 2 to x[0] and x = ", x, "\n")
   if(b and boolTest(x)):
      print("b and booltest(x) are both true.", "\n")
      x[0] += 2
      print("In main(), just added 2 to x[0] and x = ", x, "\n")
   if(b or boolTest(x)):
      print("b or booltest(x) is true.", "\n")
      x[0] += 2
      print("In main(), just added 2 to x[0] and x = ", x, "\n")
   print("End of main() and x = ", x, "\n") 

def boolTest(z [int]) bool:
   print("In boolTest() and z = ", z, "\n")
   z[0] += 10
   print("In boolTest(), just added 10 to z[0] and z = ", z, "\n")
   return true
