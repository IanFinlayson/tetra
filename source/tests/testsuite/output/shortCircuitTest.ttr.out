In main(), just created a and a = 1
In main(), just created b and b = 0
In main(), just created x and x = Len = 1 Elements: {0}
In main(), just created c and c =  and true
In main(), just created d and d =  or true
In boolTest() and z = Len = 1 Elements: {0}
In boolTest(), just added 10 to z[0] and z = Len = 1 Elements: {10}
a and booltest(x) are both true.
In main(), just added 2 to x[0] and x = Len = 1 Elements: {12}
a or booltest(x) is true.
In main(), just added 2 to x[0] and x = Len = 1 Elements: {14}
In boolTest() and z = Len = 1 Elements: {14}
In boolTest(), just added 10 to z[0] and z = Len = 1 Elements: {24}
b or booltest(x) is true.
In main(), just added 2 to x[0] and x = Len = 1 Elements: {26}
End of main() and x = Len = 1 Elements: {26}

